### PR TITLE
Upgrades

### PR DESCRIPTION
**Pull request by MikeBishop**
Here are the changes that I've made. If you prefer, I can split them out into separate PRs to discuss individually, but at a high level:

- Some refactoring to avoid repeating identical code in multiple places. Didn't ultimately consolidate very-similar code, though that might be worth considering in the future.
- Added vacation mode command; takes optional start/end dates and desired temperature, with sensible defaults (today only, minimum temperature)
- Added follow schedule command; no arguments
- Added refresh command to child device, mostly for ease of debugging
- Changed thermostatFanMode to be off, since this thermostat doesn't control a fan as far as I know

### CODE REVIEW

File: SchluterDitraHeatChild.groovy [hunk 1]
```diff
@@ -103,40 +112,95 @@ def setThermostatFanMode(fanmode) {
 }
 
 def auto() {
-    log.warn("auto() is not supported and takes no action.")
+    log.warn("auto() is not supported and takes no action. Consider setting the thermostat to Follow Schedule instead.")
```
Comment (marcre) [inline SchluterDitraHeatChild.groovy:115]: Nit - auto only applies to thermostats that support automatic switching between heating and cooling.  That's a bit different than following the schedule or not.

**Review by marcre — APPROVED** [review]
Looks good with one tiny nit

File: SchluterDitraHeatChild.groovy (hunk 1 above)
Comment (MikeBishop) [reply SchluterDitraHeatChild.groovy:115]: Yes, which is why it's no longer the same function. However, in the context of `auto()` (do something) versus `off()` (do nothing), I think it's likely someone calling `auto()` is looking to get the thermostat active again. Suggesting to resume schedule seems like the most sensible next step to me.

That said, the hints in the warnings aren't at all critical; if you want this one gone, no worries.

Comment (marcre) [reply SchluterDitraHeatChild.groovy:115]: I get what you are aiming for, might tweak the messages more, but no reason to block the PR.

**Pull request merged**